pkg/executables/cmk.go

```diff
@@ -0,0 +1,308 @@
+package executables
+
+import (
+	"bytes"
+	"context"
+	_ "embed"
+	"encoding/json"
+	"fmt"
+	"path/filepath"
+	"strconv"
+	"time"
+
+	"github.com/aws/eks-anywhere/pkg/filewriter"
+	"github.com/aws/eks-anywhere/pkg/logger"
+	"github.com/aws/eks-anywhere/pkg/retrier"
+	"github.com/aws/eks-anywhere/pkg/templater"
+)
+
+//go:embed config/cmk.ini
+var cmkConfigTemplate string
+
+const (
+	cmkConfigFileName = "cmk_tmp.ini"
+	maxRetriesCmk     = 5
+	backOffPeriodCmk  = 5 * time.Second
+)
+
+type Cmk struct {
+	writer     filewriter.FileWriter
+	executable Executable
+	retrier    *retrier.Retrier
+}
+
+// TODO: Add support for domain, account filtering
+func (c *Cmk) ValidateTemplatePresent(ctx context.Context, config CmkExecConfig, domain, zone, account, template string) error {
+	result, err := c.execWithNameAndIdFilters(ctx, config, template, "list", "templates", "templatefilter=all", "listall=true")
+	if err != nil {
+		return fmt.Errorf("error getting templates info: %v", err)
+	}
+	if result.Len() == 0 {
+		return fmt.Errorf("template %s not found", template)
+	}
+
+	response := struct {
+		CmkTemplates []CmkTemplate `json:"template"`
+	}{}
+	err = json.Unmarshal(result.Bytes(), &response)
+	if err != nil {
```

Comment:
nit: you can do
```suggestion
    if err = json.Unmarshal(result.Bytes(), &response); err != nil {
```

Reply:
I'll make these one-liners wherever possible - thanks for the hint